ledger/store/src/helpers/rocksdb/internal/nested_map.rs

```diff
@@ -0,0 +1,1890 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(clippy::type_complexity)]
+
+use super::*;
+use crate::helpers::{NestedMap, NestedMapRead};
+
+use core::{fmt, fmt::Debug, hash::Hash, mem};
+use std::{borrow::Cow, sync::atomic::Ordering};
+use tracing::error;
+
+pub const SEPARATOR: [u8; 5] = [0xFF, 0x00, 0xFF, 0x00, 0xFF];
```

Comment:
what's the significance of this separator? is it just a variant of `0xDEADBEEF`?

Reply:
but then again, as with the in-memory variant, do we need a separator here?

Reply:
ok, now that I see how it's used I understand the choice of bytes; I wonder if it wouldn't be better for us to instead reserve the very first `DataID` and then have a plain `0u16` as the separator?

Reply:
hmm, but this could cause issues with the iterators searching for the separator :thinking: